RELEASE_NOTES.md

```diff
@@ -1,6 +1,23 @@
 Release Notes
 =============
 
+v1.0.0-alpha.x
+--------------
+
+### New features
+
+- Variables are now expanded in any string properties of an entity's
+  traits. Expansion occurs during resolve to allow for variation in the
+  environment between calls. Note: Only the `$var` and `${var}` syntax
+  is supported. You can escape a literal `$` using `$$`. In addition to
+  the environment, BAL also provides several implicit variables of its
+  own:
+  - `bal_library_path` The absolute path to the current library.
```

Comment:
[consider][minor] Perhaps `current library file` to make it clear that this is going to be a file path, as I can imagine some people thinking library == directory structure.

Reply:
Do you think its still ambiguous in the broader context where your entire interaction with BAL is in relation to a library (singular), in a JSON file? If thats a concern, we'd probably need to tackle the ambiguity elsewhere too?

My only thought on 'current' is that it implies there may have been others? 

I quite liked the correlation with the (optionally used) `BAL_LIBRARY_PATH` env var too.

Reply:
It's only ambiguous to a non BAL user, ie someone who encounters BAL as a tertiary component in something else they're doing. The moment you see that the library means a file the ambiguity is gone.

forget about `current`, as usual, not thinking when i type, i mean something more like `bal_library_file_path`, just to get the idea that this is a file in the name.

Reply:
Cool, thanks. Shall we leave this as-is for now, as it seems like we'd want to tackle that with the env var too if we did change anything.